test/test_storage.py

```diff
@@ -169,6 +169,22 @@ def test_save_notice(self):
             list(store.notices('event')),
             [('event', 'observer', 'method')])
 
+    def test_all_notices(self):
+        notices = [('e1', 'o1', 'm1'), ('e1', 'o2', 'm2'), ('e2', 'o3', 'm3')]
+        store = self.create_storage()
+        for notice in notices:
+            store.save_notice(*notice)
+
+        for arg, expected in [
+                ('e1', notices[:2]),
+                ('e2', notices[2:]),
+                (None, notices),
+                ('', notices),
```

Comment:
this is the `notices('')` case. Should I make it more explicit? Probably more readable even if it's a lot of (one-line) repetition